### DISCUSSION TITLE
Naming conventions: "libs" and "apps" vs "packages" (vs none)

### DISCUSSION THREAD

**Discussion by masonmark** [nrwl/nx#16184] [General]
What is the setting that controls the name of the subfolder that libraries live in? At the monorepo I use at work, it is "libs" and our apps are in "apps". However, recently when I create a new monorepo the folder for both libs and apps is most often "packages". Or sometimes, there seems to be no such setting and new libraries are just generated as subfolders in the monorepo root.

I think this depends on how the monorepo is created initially.

I am pretty sure there is a setting to control this, because I asked this same question about a year ago on the Nx Slack, and somebody did kindly tell me the name of the setting. Unfortunately Slack can't search conversations older than three months, so I thought I'd ask again here.

I think I'd like to change this setting, so my personal monorepos are set up similarly to the one I use at work. However, I would also like to know if the thinking has changed lately, and separating things into "libs" and "apps" is no longer the recommendation.

**Comment by rolginroman** [comment]
Hello, @masonmark 

You could use `nx.json` configuration for that

```
  "workspaceLayout": {
    "appsDir": "packages",
    "libsDir": "libs"
  }, 
```

[schema.json link](https://github.com/nrwl/nx/blob/master/packages/nx/schemas/nx-schema.json#L46)

However, nx will find any project by `project.json` configuration file or even with any other markers that might be provided with external plugins for other languages

**Comment by tamtakoe** [comment]
Can I set `appsDir` by command line (not nx.json)?
E.g. `nx create-nx-workspace app --appsDir="apps"` ?

And why we need this options (appsDir, libsDir) when I can create apps/libs dir by `nx g @nx/nest:app apps/my-nest-app` or `nx g @nx/nest:app my-nest-app --directory="apps/my-nest-app"`?
